request_handler/request_handler.go

```diff
@@ -189,6 +212,23 @@ func (s *SpadeHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		w.WriteHeader(http.StatusBadRequest)
 		return
 	}
+
+	origin := r.Header.Get("Origin")
+	if s.corsOrigins[origin] {
+		w.Header().Set("Access-Control-Allow-Origin", origin)
+		w.Header().Set("Access-Control-Allow-Methods", "GET, POST, OPTIONS")
+	}
+
+	if r.Method == "OPTIONS" {
+		w.Header().Set("Access-Control-Max-Age", corsMaxAge)
+		w.WriteHeader(http.StatusOK)
+		return
+	}
+
+	if r.Method == "GET" {
+		w.Header().Set("Vary", "Origin")
+	}
```

Comment:
:sheep: I think the Go way for this would be
```go
switch (r.Method) {
case "OPTIONS":
    // do OPTIONS block from above
case "GET":
    // do GET block from above
}